### ISSUE TITLE
React.pureComponent ignored (React v15.3.0)

### ISSUE THREAD

**Issue by ozum** [yannickcr/eslint-plugin-react#737]
Using eslint-plugin-react v6.0.0 `"react/prop-types": ["error"]` rule does not generate error for `React.PureComponent` which is added in react v15.3.0.

Example:
`"react/prop-types": ["error"]`

```js
import React from 'react';

export default class extends React.PureComponent {
  render() {
    return (
      <div>
        <span>{this.props.no}</span>
      </div>
    );
  }
}
```

**Comment by lencioni** [comment]
Thanks for the report! I think we need to update some of our utility methods to better detect these new things. e.g. https://github.com/yannickcr/eslint-plugin-react/blob/763382f8d4ec5961c765bbb71cd2698016da5b53/lib/util/Components.js#L195

@ozum would you be interested in submitting a PR for this?

**Comment by ozum** [comment]
Unfortunately,I don't have enough React experience for this PR. This is my first week with React.

**Comment by yannickcr** [comment]
Working on it
